crates/ide-completion/src/completions/expr.rs

```diff
@@ -9,6 +13,39 @@ use crate::{
     CompletionContext, Completions,
 };
 
+struct PathCallback<'a, F> {
+    ctx: &'a CompletionContext<'a>,
+    acc: &'a mut Completions,
+    add_assoc_item: F,
+    seen: FxHashSet<hir::AssocItem>,
```

Comment:
how come we need to have this now, we didn't before

Reply:
Oh you are right! Must have duplicated this from other places without thinking. Will remove that.

Reply:
Oh actually we did have this before, it was just hidden behind `iterate_path_candidates()`, but now we cannot use this method because we need to tell between inherent and trait methods.

Reply:
Oh actually we did have this before, it was just hidden behind `iterate_path_candidates()`, but now we cannot use this method because we need to tell between inherent and trait methods.